ios/BVLinearGradientLayer.m

```diff
@@ -60,15 +60,32 @@ - (void)display {
         hasAlpha = hasAlpha || CGColorGetAlpha(self.colors[i].CGColor) < 1.0;
     }
 
-    UIGraphicsBeginImageContextWithOptions(self.bounds.size, !hasAlpha, 0.0);
-    CGContextRef ref = UIGraphicsGetCurrentContext();
-    [self drawInContext:ref];
+    if (@available(iOS 10.0, *)) {
+        UIGraphicsImageRendererFormat *format;
+        if (@available(iOS 11.0, *)) {
+            format = [UIGraphicsImageRendererFormat preferredFormat];
+        } else {
+            format = [UIGraphicsImageRendererFormat defaultFormat];
+        }
+        format.opaque = !hasAlpha;
```

Comment:
Previously the opaque field wasn't being set for the iOS 10+ flow. I think by default it's `false` with no `format` passed in.

Looking at the docs, [`defaultFormat`](https://developer.apple.com/documentation/uikit/uigraphicsrendererformat/1648550-default) seems like a decent option, but it points to [`preferredFormat`](https://developer.apple.com/documentation/uikit/uigraphicsrendererformat/2928214-preferred) for immediate rendering so I think that one should be the main choice. Though it's only available in iOS 11+ so that available check needs to be done again 😅.

Reply:
Sounds good, thanks for the explanation!

Reply:
Can we publish a new version prior to 3.0.0 with this change ? .. iOS17 is around the corner

Reply:
@iit2008047 - Sure thing! It looks like the patch applies cleanly to the 2.8 branch. I'll cherry-pick it shortly and release the next 2.8 patch version (it will be 2.8.3).